programs/marinade-finance/src/instructions/user/withdraw_stake_account.rs

```diff
@@ -0,0 +1,302 @@
+use crate::{
+    checks::check_token_source_account,
+    error::MarinadeError,
+    events::user::WithdrawStakeAccountEvent,
+    state::{stake_system::StakeSystem, validator_system::ValidatorSystem},
+    State,
+};
+
+use anchor_lang::solana_program::{native_token::LAMPORTS_PER_SOL, program::invoke_signed, stake};
+use anchor_lang::{
+    prelude::*,
+    solana_program::{program::invoke, stake::state::StakeAuthorize},
+};
+use anchor_spl::{
+    stake::{
+        deactivate_stake as solana_deactivate_stake, DeactivateStake as SolanaDeactivateStake,
+        Stake, StakeAccount,
+    },
+    token::{burn, Burn, Mint, Token, TokenAccount},
+};
+
+use crate::checks::check_stake_amount_and_validator;
+
+#[derive(Accounts)]
+pub struct WithdrawStakeAccount<'info> {
+    #[account(mut)]
+    pub state: Box<Account<'info, State>>,
+
+    #[account(mut)]
+    pub msol_mint: Box<Account<'info, Mint>>,
+
+    // Note: new stake account withdraw-auth (owner) & staker-auth will be owner of burn_msol_from
+    #[account(
+        mut,
+        token::mint = state.msol_mint
```

Comment:
don't think this works. `token::mint = msol_mint` must be what you are trying to achieve

Reply:
it works, but no problem changing it